```
[ 18%] Building CXX object Modules/Core/CMakeFiles/MitkCore.dir/src/DataManagement/mitkEnumerationProperty.cpp.o
In file included from /usr/lib/gcc/x86_64-pc-linux-gnu/4.9.3/include/g++-v4/shared_mutex:35:0,
                 from /var/lib/jenkins/home/jobs/MITK-PR/workspace/Modules/Core/src/DataManagement/mitkEnumerationProperty.cpp:19:
/usr/lib/gcc/x86_64-pc-linux-gnu/4.9.3/include/g++-v4/bits/c++14_warning.h:32:2: error: #error This file requires compiler and library support for the forthcoming ISO C++ 2014 standard. This support is currently experimental, and must be enabled with the -std=c++1y or -std=gnu++1y compiler options.
 #error This file requires compiler and library support for the forthcoming \
  ^
/var/lib/jenkins/home/jobs/MITK-PR/workspace/Modules/Core/src/DataManagement/mitkEnumerationProperty.cpp:24:6: error: ‘shared_mutex’ in namespace ‘std’ does not name a type
 std::shared_mutex s_IdMapForClassNameGuard;
      ^
/var/lib/jenkins/home/jobs/MITK-PR/workspace/Modules/Core/src/DataManagement/mitkEnumerationProperty.cpp:25:6: error: ‘shared_mutex’ in namespace ‘std’ does not name a type
 std::shared_mutex s_StringMapForClassNameGuard;
```

интересно на 5.4 соберётся или нет

--- a/Modules/Core/src/DataManagement/mitkEnumerationProperty.cpp
+++ b/Modules/Core/src/DataManagement/mitkEnumerationProperty.cpp
@@ -16,13 +16,14 @@ See LICENSE.txt or http://www.mitk.org for details.
 
 #include "mitkEnumerationProperty.h"
 #include <algorithm>
-#include <shared_mutex>
+#include <boost/thread/locks.hpp>
+#include <boost/thread/shared_mutex.hpp>
 
 // static map members of EnumerationProperty. These Maps point to per-classname-maps of ID <-> String. Accessed by GetEnumIds() and GetEnumString().
 mitk::EnumerationProperty::IdMapForClassNameContainerType     mitk::EnumerationProperty::s_IdMapForClassName;
 mitk::EnumerationProperty::StringMapForClassNameContainerType mitk::EnumerationProperty::s_StringMapForClassName;
-std::shared_mutex s_IdMapForClassNameGuard;
-std::shared_mutex s_StringMapForClassNameGuard;
+boost::shared_mutex s_IdMapForClassNameGuard;
+boost::shared_mutex s_StringMapForClassNameGuard;
 
 mitk::EnumerationProperty::EnumerationProperty()
 {
@@ -177,13 +178,13 @@ mitk::EnumerationProperty::EnumIdsContainerType& mitk::EnumerationProperty::GetE
 {
   std::string className = this->GetNameOfClass(); // virtual!
   {
-    std::shared_lock<std::shared_mutex> lock(s_IdMapForClassNameGuard);
+    boost::shared_lock<boost::shared_mutex> lock(s_IdMapForClassNameGuard);
     auto iter = s_IdMapForClassName.find(className);
     if (s_IdMapForClassName.end() != iter) {
       return iter->second;
     }
   }
-  std::unique_lock<std::shared_mutex> lock(s_IdMapForClassNameGuard);
+  boost::unique_lock<boost::shared_mutex> lock(s_IdMapForClassNameGuard);
   return s_IdMapForClassName[className];
 }
 
@@ -192,13 +193,13 @@ const mitk::EnumerationProperty::EnumIdsContainerType& mitk::EnumerationProperty
 {
   std::string className = this->GetNameOfClass(); // virtual!
   {
-    std::shared_lock<std::shared_mutex> lock(s_IdMapForClassNameGuard);
+    boost::shared_lock<boost::shared_mutex> lock(s_IdMapForClassNameGuard);
     auto iter = s_IdMapForClassName.find(className);
     if (s_IdMapForClassName.end() != iter) {
       return iter->second;
     }
   }
-  std::unique_lock<std::shared_mutex> lock(s_IdMapForClassNameGuard);
+  boost::unique_lock<boost::shared_mutex> lock(s_IdMapForClassNameGuard);
   return s_IdMapForClassName[className];
 }
 
@@ -207,13 +208,13 @@ mitk::EnumerationProperty::EnumStringsContainerType& mitk::EnumerationProperty::
 {
   std::string className = this->GetNameOfClass(); // virtual!
   {
-    std::shared_lock<std::shared_mutex> lock(s_StringMapForClassNameGuard);
+    boost::shared_lock<boost::shared_mutex> lock(s_StringMapForClassNameGuard);
     auto iter = s_StringMapForClassName.find(className);
     if (s_StringMapForClassName.end() != iter) {
       return iter->second;
     }
   }
-  std::unique_lock<std::shared_mutex> lock(s_StringMapForClassNameGuard);
+  boost::unique_lock<boost::shared_mutex> lock(s_StringMapForClassNameGuard);
   return s_StringMapForClassName[className];
 }
 
@@ -222,13 +223,13 @@ const mitk::EnumerationProperty::EnumStringsContainerType& mitk::EnumerationProp
 {
   std::string className = this->GetNameOfClass(); // virtual!
   {
-    std::shared_lock<std::shared_mutex> lock(s_StringMapForClassNameGuard);
+    boost::shared_lock<boost::shared_mutex> lock(s_StringMapForClassNameGuard);
     auto iter = s_StringMapForClassName.find(className);
     if (s_StringMapForClassName.end() != iter) {
       return iter->second;
     }
   }
-  std::unique_lock<std::shared_mutex> lock(s_StringMapForClassNameGuard);
+  boost::unique_lock<boost::shared_mutex> lock(s_StringMapForClassNameGuard);
   return s_StringMapForClassName[className];
 }
 

